diff --git a/astroid/interpreter/objectmodel.py b/astroid/interpreter/objectmodel.py
index fd8c0c0..8ff7d50 100644
--- a/astroid/interpreter/objectmodel.py
+++ b/astroid/interpreter/objectmodel.py
@@ -924,12 +924,9 @@ class PropertyModel(ObjectModel):
                 nonlocal func
                 if caller and len(caller.args) != 1:
                     raise InferenceError(
-                        "fget() needs a single argument", target=self, context=context
+                        "fget() needs one argument", target=self, context=context
                     )
-
-                yield from func.function.infer_call_result(
-                    caller=caller, context=context
-                )
+                yield from func.function.infer_call_result(caller=caller, context=context)
 
         property_accessor = PropertyFuncAccessor(
             name="fget",
@@ -939,9 +936,8 @@ class PropertyModel(ObjectModel):
             end_lineno=self._instance.end_lineno,
             end_col_offset=self._instance.end_col_offset,
         )
-        property_accessor.postinit(args=func.args, body=func.body)
+        property_accessor.postinit(args=func.function.args, body=func.function.body)
         return property_accessor
-
     @property
     def attr_fset(self):
         func = self._instance
