diff --git a/astroid/nodes/node_ng.py b/astroid/nodes/node_ng.py
index c6a044f..1e9772b 100644
--- a/astroid/nodes/node_ng.py
+++ b/astroid/nodes/node_ng.py
@@ -380,17 +380,13 @@ class NodeNG:
         """
         for field in self._astroid_fields:
             node_or_sequence = getattr(self, field)
-            # /!\ compiler.ast Nodes have an __iter__ walking over child nodes
-            if child is node_or_sequence:
-                return field, child
-            if (
-                isinstance(node_or_sequence, (tuple, list))
-                and child in node_or_sequence
-            ):
+            if node_or_sequence is child:
+                return field, node_or_sequence
+            if isinstance(node_or_sequence, (list, tuple)) and child in node_or_sequence:
                 return field, node_or_sequence
+
         msg = "Could not find %s in %s's children"
         raise AstroidError(msg % (repr(child), repr(self)))
-
     # FIXME : should we merge child_sequence and locate_child ? locate_child
     # is only used in are_exclusive, child_sequence one time in pylint.
 
