diff --git a/astroid/constraint.py b/astroid/constraint.py
index 08bb80e..2422c6e 100644
--- a/astroid/constraint.py
+++ b/astroid/constraint.py
@@ -130,11 +130,10 @@ def _matches(node1: nodes.NodeNG | bases.Proxy, node2: nodes.NodeNG) -> bool:
     return False
 
 
-def _match_constraint(
-    node: _NameNodes, expr: nodes.NodeNG, invert: bool = False
-) -> Iterator[Constraint]:
+def _match_constraint(node: _NameNodes, expr: nodes.NodeNG, invert: bool=False
+    ) -> Iterator[Constraint]:
     """Yields all constraint patterns for node that match."""
-    for constraint_cls in ALL_CONSTRAINT_CLASSES:
-        constraint = constraint_cls.match(node, expr, invert)
-        if constraint:
-            yield constraint
+    for constraint_class in ALL_CONSTRAINT_CLASSES:
+        constraint = constraint_class.match(node, expr, negate=invert)
+        if constraint is not None:
+            yield constraint
\ No newline at end of file
